sse/src/main/scala/io/finch/sse/ServerSentEvent.scala

```diff
@@ -0,0 +1,51 @@
+package io.finch.sse
+
+import java.nio.charset.Charset
+
+import cats.Show
+import com.twitter.concurrent.AsyncStream
+import com.twitter.io.Buf
+import io.finch.{Encode, Ok, Output, Text}
+import io.finch.internal.{BufText, LowPriorityToResponseInstances}
+
+case class ServerSentEvent[A](
+  data: A,
+  id: Option[String] = None,
+  event: Option[String] = None,
+  retry: Option[Long] = None
+)
+
+object ServerSentEvent extends LowPriorityToResponseInstances {
+  type EventStream[A] = AsyncStream[ServerSentEvent[A]]
+
+  def stream[A](sse: EventStream[A]): Output[EventStream[A]] = {
+    Ok(sse).withHeader("Cache-Control" -> "no-cache").withHeader("Connection" -> "keep-alive")
+  }
+
+  implicit def sseAsyncToResponse[A](implicit
+     e: Encode.Aux[A, Text.EventStream]
+  ): Aux[AsyncStream[A], Text.EventStream] =
+    asyncResponseBuilder((a, cs) => e(a, cs).concat(Buf.Utf8("\n")))
+
+  implicit def encodeEventStream[A](implicit s: Show[A]): Encode.Aux[ServerSentEvent[A], Text.EventStream] = {
+    Encode.instance[ServerSentEvent[A], Text.EventStream]({ (event: ServerSentEvent[A], c: Charset) =>
+      encodeEvent[A](event, c, { (a: A, cs: Charset) => BufText(s.show(a), cs) })
+    })
+  }
+
+  def encodeEvent[A](sse: ServerSentEvent[A], cs: Charset, e: (A, Charset) => Buf): Buf = {
+    val dataBuf = BufText("data:", cs).concat(e(sse.data, cs)).concat(BufText("\n", cs))
+    val eventType = sse.event.map(e => s"event:$e\n").getOrElse("")
+    val id = sse.id.map(id => s"id:$id\n").getOrElse("")
+    val retry = sse.retry.map(retry => s"retry:$retry\n").getOrElse("")
+    val restBuf = BufText(eventType + id + retry, cs)
+    dataBuf.concat(restBuf)
+  }
+
+  /** By default Finch encodes an exception as a Comment */
+  implicit val encodeSseException: Encode.Aux[Exception, Text.EventStream] = {
```

Comment:
It does, but its not great. The server does produce a new event, but it looks like browsers discard comments. We might want to rethink how we treat errors.